app/src/main/java/org/onionshare/android/tor/TorManager.kt

```diff
@@ -66,10 +66,14 @@ class TorManager @Inject constructor(
                     LOG.debug("TorService: Started")
                     startLatch?.countDown() ?: LOG.error("startLatch was null when Tor started.")
                 }
-                // FIXME When we stop unplanned, we need to inform the ShareManager
-                //  that we stopped, so it can clear its state up, stopping webserver, etc.
-                TorService.STATUS_STOPPING -> LOG.debug("TorService: Stopping...")
-                TorService.STATUS_OFF -> LOG.debug("TorService: Stopped")
+                TorService.STATUS_STOPPING -> {
+                    LOG.debug("TorService: Stopping...")
+                    _state.value = TorState.Stopping
```

Comment:
Looks sensible to reflect the status in our own StateFlow. Should we do the same for STATUS_STARTING and STATUS_ON?

Reply:
I prefer to set STARTING earlier, so as soon as we called `start()` our state is STARTING already.

Here, STATUS_ON, counts down the startLatch, so we can create the HS and only once that has been published, we consider ourselves STARTED, so there's some divergence between the TorService states and the states we use for our own state management.